packages/observable-client/src/chainHead/chainHead.ts

```diff
@@ -407,34 +407,46 @@ export const getChainHead$ = (chainHead: ChainHead) => {
     () => of(),
   )
 
-  const genesis$ = runtime$.pipe(
-    filter(Boolean),
+  const genesis$ = pinnedBlocks$.pipe(
+    filter(({ blocks }) => blocks.size > 0),
     take(1),
-    mergeMap((runtime) => {
-      const { enc } = runtime.dynamicBuilder.buildStorage(
-        "System",
-        "BlockHash",
-      ).keys
-      // const genesis$ =
-      // there are chains (e.g. kilt) that use u64 as block number
-      // u64 is encoded as bigint
-      // using dynamic builder for safety
-      let key: string
-      try {
-        // for u32
-        key = enc(0)
-      } catch {
-        // for u64
-        key = enc(0n)
-      }
-
-      return storage$(
-        null,
-        false,
-        "value",
-        () => key,
-        null,
-      ) as Observable<HexString>
+    mergeMap(({ blocks }) => {
+      const hash = Array.from(blocks.values()).find(
+        ({ number }) => number === 0,
+      )?.hash
+      return hash
+        ? of(hash)
+        : runtime$.pipe(
+            filter(Boolean),
+            take(1),
+            mergeMap((runtime) => {
+              if (typeof runtime === "string") return runtime
```

Comment:
This case is impossible, right? runtime shouldn't be a string

Reply:
yep, previous impl haha